File: Server/src/common/infrastructure/cryptoAlgorithm/CipherEncryption.cpp
#include "CipherEncryption.h"

std::string CipherEncryption::encrypt(const std::string &message) const
{
    std::string encrypted = "";
    for (const char c : message)
    {
        encrypted += static_cast<char>(c + key);
    }

    return encrypted;
}

std::string CipherEncryption::decrypt(const std::string &message) const
{
    std::string decrypted = "";
    for (const char c : message)
    {
        decrypted += static_cast<char>(c - key);
    }

    return decrypted;
}

File: Client/Trivia/Source/CryptoAlgorithm/CipherEncryption.cs
using System.Linq;

namespace Trivia.CryptoAlgorithm;

public class CipherEncryption : ICryptoAlgorithm
{
    private const int Key = 3;

    public string Encrypt(string message)
    {
        return message
            .Aggregate("", (current, c) =>
                current + (char)(c + Key)
            );
    }

    public string Decrypt(string message)
    {
        return message
            .Aggregate("", (current, c) =>
                current + (char)(c - Key)
            );
    }
}